### ISSUE TITLE
Mystery hang during /usr/bin/kubeadm init (with no --extra-config options)

### ISSUE THREAD

**Issue by hsebastian** [kubernetes/minikube#2939]
BUG REPORT

**Environment**:

```
$ minikube version
minikube version: v0.28.0

$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

$ grep DriverName ~/.minikube/machines/minikube/config.json
    "DriverName": "virtualbox",

$ grep -i ISO ~/.minikube/machines/minikube/config.json
        "Boot2DockerURL": "file:///home/hans/.minikube/cache/iso/minikube-v0.28.0.iso",

$ cat /home/hans/.kube/config
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /home/hans/.minikube/ca.crt
    server: https://192.168.99.103:8443
  name: minikube
contexts:
- context:
    cluster: minikube
    user: minikube
  name: minikube
current-context: minikube
kind: Config
preferences: {}
users:
- name: minikube
  user:
    client-certificate: /home/hans/.minikube/client.crt
    client-key: /home/hans/.minikube/client.key

```

**What happened**: When running `minikube start` it hangs at `Starting cluster components...`. In the console of the minikube VM, I get prompted for `minikube login: `. I can login with `root`.

**What you expected to happen**: minikube is started and configured.

**How to reproduce it** (as minimally and precisely as possible):
```
minikube delete
minikube start --logtostderr --loglevel 0
```
**Output of `minikube logs` (if applicable)**:
```
Connecting to cluster...
Setting up kubeconfig...
I0627 11:33:56.307740   24241 config.go:101] Using kubeconfig:  /home/hans/.kube/config
Starting cluster components...
I0627 11:33:56.308563   24241 ssh_runner.go:80] Run with output: sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI 
```
**Anything else do we need to know**:
Installed minikube with
```
curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
```
Installed virtualbox with
```
sudo apt-get remove --purge virtualbox-5.2
sudo apt-get remove --purge virtualbox
rm -rf ~/.config/VirtualBox
sudo dpkg -i ~/Downloads/virtualbox-5.2_5.2.12-122591\~Ubuntu\~xenial_amd64.deb
```

**Comment by imroc** [comment]
I got the same issue

**Comment by hsebastian** [comment]
@Arnavion No, I remember it was 1.10.0 and I also tried with 1.9.4 but still fails. I didn't know there is 1.11.0.

**Comment by IBMRob** [comment]
We are also hitting this, it looks the same as we were hitting previously as well but the issue was closed - https://github.com/kubernetes/minikube/issues/2765

For now some of us are using 0.25.2 OK.
Others have problem with that version which need a newer docker version to pick up a fix so are completely blocked as all newer versions hit this problem

**Comment by Karreg** [comment]
Same here for Minikube 0.28.2.
It happens with both VirtualBox and Hyper-V drivers

**Comment by exu** [comment]
Same here but when starting with logs I've got multiple hundreds of messages like this:

```
minikube start --logtostderr --loglevel 0                                                     
```

```
I0810 10:49:23.576654    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:24.076413    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:24.576264    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:25.076410    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:25.576426    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:26.076719    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:26.577313    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:27.077177    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:27.577812    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:28.078182    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:28.577055    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: dial tcp 192.168.99.100:8443: getsockopt: connection refused]
I0810 10:49:39.076599    8694 kubernetes.go:119] error getting Pods with label selector "k8s-app=kube-proxy" [Get https://192.168.99.100:8443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-proxy: net/http: TLS handshake timeout]
I0810 10:49:43.551134    8694 kubernetes.go:124] Found 0 Pods for label selector k8s-app=kube-proxy
```
and it finally hangs and not displaying anything

**Comment by frastel** [comment]
I'm also not able to boot a Kubernetes 1.10 Cluster with a Minikube v0.28.2 on a ubuntu xenial vm with "non" driver/localkube.

After one hour I got this output

```
==> default: Starting local Kubernetes v1.10.0 cluster...
==> default: Starting VM...
==> default: Getting VM IP address...
==> default: Moving files into cluster...
==> default: Downloading kubeadm v1.10.0
==> default: Downloading kubelet v1.10.0
==> default: Finished Downloading kubelet v1.10.0
==> default: Finished Downloading kubeadm v1.10.0
==> default: Setting up certs...
==> default: Connecting to cluster...
==> default: Setting up kubeconfig...
==> default: Starting cluster components...
==> default: E0810 08:31:32.155043    4024 start.go:300] Error starting cluster:  kubeadm init error 
==> default: sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
==> default: sudo /usr/bin/kubeadm alpha phase addon kube-dns
==> default:  running command: : running command: 
==> default: sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
==> default: sudo /usr/bin/kubeadm alpha phase addon kube-dns
==> default: 
==> default:  output: [init] Using Kubernetes version: v1.10.0
==> default: [init] Using Authorization modes: [Node RBAC]
==> default: [preflight] Running pre-flight checks.
==> default: 	[WARNING SystemVerification]: docker version is greater than the most recently validated version. Docker version: 17.09.1-ce. Max validated version: 17.03
==> default: 	[WARNING Hostname]: hostname "minikube" could not be reached
==> default: 	[WARNING Hostname]: hostname "minikube" lookup minikube on 10.0.2.3:53: no such host
==> default: 	[WARNING FileExisting-ebtables]: ebtables not found in system path
==> default: 	[WARNING FileExisting-socat]: socat not found in system path
==> default: 	[WARNING FileExisting-crictl]: crictl not found in system path
==> default: Suggestion: go get github.com/kubernetes-incubator/cri-tools/cmd/crictl
==> default: Flag --admission-control has been deprecated, Use --enable-admission-plugins or --disable-admission-plugins instead. Will be removed in a future version.
==> default: 	[WARNING ExtraArgs]: kube-apiserver: failed to parse extra argument --ServiceNodePortRange=80-32767
==> default: [certificates] Using the existing ca certificate and key.
==> default: [certificates] Using the existing apiserver certificate and key.
==> default: [certificates] Generated apiserver-kubelet-client certificate and key.
==> default: [certificates] Generated etcd/ca certificate and key.
==> default: [certificates] Generated etcd/server certificate and key.
==> default: [certificates] etcd/server serving cert is signed for DNS names [localhost] and IPs [127.0.0.1]
==> default: [certificates] Generated etcd/peer certificate and key.
==> default: [certificates] etcd/peer serving cert is signed for DNS names [minikube] and IPs [10.0.2.15]
==> default: [certificates] Generated etcd/healthcheck-client certificate and key.
==> default: [certificates] Generated apiserver-etcd-client certificate and key.
==> default: [certificates] Generated sa key and public key.
==> default: [certificates] Generated front-proxy-ca certificate and key.
==> default: [certificates] Generated front-proxy-client certificate and key.
==> default: [certificates] Valid certificates and keys now exist in "/var/lib/localkube/certs/"
==> default: [kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
==> default: [kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
==> default: [kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
==> default: [kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
==> default: [controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/manifests/kube-apiserver.yaml"
==> default: [controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/manifests/kube-controller-manager.yaml"
==> default: [controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/manifests/kube-scheduler.yaml"
==> default: [etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
==> default: [init] Waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests".
==> default: [init] This might take a minute or longer if the control plane images have to be pulled.
==> default: 
==> default: Unfortunately, an error has occurred:
==> default: 	timed out waiting for the condition
==> default: 
==> default: This error is likely caused by:
==> default: 	- The kubelet is not running
==> default: 	- The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)
==> default: 	- Either there is no internet connection, or imagePullPolicy is set to "Never",
==> default: 	  so the kubelet cannot pull or find the following control plane images:
==> default: 		- k8s.gcr.io/kube-apiserver-amd64:v1.10.0
==> default: 		- k8s.gcr.io/kube-controller-manager-amd64:v1.10.0
==> default: 		- k8s.gcr.io/kube-scheduler-amd64:v1.10.0
==> default: 		- k8s.gcr.io/etcd-amd64:3.1.12 (only if no external etcd endpoints are configured)
==> default: 
==> default: If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
==> default: 	- 'systemctl status kubelet'
==> default: 	- 'journalctl -xeu kubelet'
==> default: couldn't initialize a Kubernetes cluster
==> default: : running command: 
==> default: sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
==> default: sudo /usr/bin/kubeadm alpha phase addon kube-dns
==> default: 
==> default: .: exit status 1
```

With pinned Minikube version to v0.25.2 everything works just fine.

**Comment by frastel** [comment]
To reproduce this issue I have created a [simple vagrant environment](https://github.com/chefkoch-dev/kubernetes-setup/tree/ubuntu/ubuntu/minikube).

**Comment by hsebastian** [comment]
FYI, this problem went away for me a few days ago after upgrading Virtualbox to `Version 5.2.16 r123759 (Qt5.6.1)`.

**Comment by frastel** [comment]
Mmh, can't confirm this for my setup. Virtualbox `5.2.16 r123759 (Qt5.6.3)` and Vagrant `2.1.2` installed.
I don't think the Virtualbox version has something to do with my setup as I am using localkube.

**Comment by Misoder** [comment]
I'm also experiencing this on MacOS Sierra with minikube v0.28.2 trying to start a Kubernetes v1.10.0 cluster. I've tried with hyperkit as well as virtualbox with the same outcome that @frastel posted above. Process exiting with status 1 after sudo /usr/bin/kubeadm alpha phase addon kube-dns

**Comment by imjoey** [comment]
The same issue on on MacOS High Sierra 10.13.6 with minikube v0.28.2 and the driver of virtualbox v5.1.20 for kubernetes v1.10.0.

```
Starting local Kubernetes v1.10.0 cluster...
Starting VM...
Downloading Minikube ISO
 160.27 MB / 160.27 MB [============================================] 100.00% 0s
Getting VM IP address...
Moving files into cluster...
Downloading kubeadm v1.10.0
Downloading kubelet v1.10.0
Finished Downloading kubeadm v1.10.0
Finished Downloading kubelet v1.10.0
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...

E0824 13:13:57.554917   80481 start.go:300] Error starting cluster:  kubeadm init error
sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
sudo /usr/bin/kubeadm alpha phase addon kube-dns
 running command: : running command:
sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
sudo /usr/bin/kubeadm alpha phase addon kube-dns

.: Process exited with status 1
```

**Comment by Custodia** [comment]
@Arnavion 

Is the only solution then to downgrade Kubernetes to an earlier version?

**Comment by dkushner** [comment]
Experiencing the same issue on Ubuntu 18.04 with the following command:

```
minikube start --vm-driver=kvm2 --extra-config=apiserver.Authorization.Mode=RBAC
```
My minikube version is `0.28.2`. As others have mentioned, I've tried specifying the kubernetes version and rolling back to previous versions after completely removing the `~/.minikube` directory. No result. 

Have to say, I can't remember the last time I was able to upgrade my minikube installation without incident.

**Comment by dkushner** [comment]
As usual, immediately after posting my addition to this issue,I somehow stumbled upon a solution. After stripping off the `--extra-config=apiserver.Authorization.Mode=RBAC` from my command line and manually deleting the KVM2 network `default` that minikube had created in a previous run, I was able to complete the cluster bootstrapping. I'm still unclear on which of these was causing the issue so I'm going to run a few trials of destroying the entire cluster and all associated KVM2 resources.

**Comment by tstromberg** [comment]
FYI, this step should timeout within 30 minutes and provide more insight into what happened.

That's no excuse for the lack of usable output in this case, however.

**Comment by tstromberg** [comment]
@roelal - Those messages sound an awful lot like #3031 - which in general seems like a networking issue, so you should follow-up there. Any chance there is a firewall or proxy installed on your machine?

**Comment by rliuyi** [comment]
Same issue here with:
MasOS: 10.14.1
minikube: 0.30.0
VM Driver: virtualbox

Finally solved this issue with minikube: 0.25.0, vm-driver: hyperkit

**Comment by tstromberg** [comment]
minikube v0.33.1 should now remove mystery hangs. You can watch the output streamed from kubeadm by using --alsologtostderr -v=1.

We'll also be dividing up the pull & run sections in the UI shortly, which should help, so I'm not closing this issue as obsolete just yet.

**Comment by tstromberg** [comment]
I believe this issue was resolved in the v1.1.0 release. Please try upgrading to the latest release of minikube, and if the same issue occurs, please re-open this bug. Thank you opening this bug report, and for your patience!
